src/channel.rs

```diff
@@ -68,7 +50,7 @@ fn format_nightly_url(date: &Date) -> Result<String> {
 }
 
 fn construct_channel_url(desc: &DistToolchainDescription) -> Result<String> {
-    let mut url = FUELUP_GH_PAGES.to_owned();
+    let mut url = format!("{}{}/gh-pages/", GITHUB_USER_CONTENT_URL, "fuelup");
```

Comment:
I'd prefer to keep `FUELUP_GH_PAGES` as a constant and refactor this to not use a mutable url.

```
fn construct_channel_url(desc: &DistToolchainDescription) -> Result<String> {
    let channel_file = match desc.name {
        DistToolchainName::Latest => {
            if let Some(date) = desc.date {
                &format!("channels/latest/channel-fuel-latest-{date}.toml")
            } else {
                CHANNEL_LATEST_FILE_NAME
            }
        }

        DistToolchainName::Nightly => {
            let mut file = "".to_string();
            if let Some(date) = desc.date {
                file.push_str(&format_nightly_url(&date)?);
                file.push('/');
            }
            file.push_str(CHANNEL_NIGHTLY_FILE_NAME);
            &file
        }
        DistToolchainName::Beta1 => CHANNEL_BETA_1_FILE_NAME,
        DistToolchainName::Beta2 => CHANNEL_BETA_2_FILE_NAME,
        DistToolchainName::Beta3 => CHANNEL_BETA_3_FILE_NAME,
        DistToolchainName::Beta4 => CHANNEL_BETA_4_FILE_NAME,
        DistToolchainName::Beta5 => CHANNEL_BETA_5_FILE_NAME,
        DistToolchainName::Devnet => CHANNEL_DEVNET_FILE_NAME,
        DistToolchainName::Testnet => CHANNEL_TESTNET_FILE_NAME,
    };
    let mut url = format!("{}{}/gh-pages/", GITHUB_USER_CONTENT_URL, "fuelup");


    Ok(format!("{FUELUP_GH_PAGES}{channel_file}"))
}

```